assets/src/edit-story/components/fontPicker/pickerContainer.js

```diff
@@ -0,0 +1,312 @@
+  min-width: 160px;
+  max-height: 355px;
+  overflow: hidden;
+  border-radius: 4px;
+  background-color: ${({ theme }) => theme.colors.fg.v1};
+  background-clip: padding-box;
+  box-shadow: 0 6px 12px ${({ theme }) => rgba(theme.colors.bg.v0, 0.175)};
+  padding: 0;
+`;
+
+const ListContainer = styled.div`
+  display: flex;
+  flex-direction: row;
+  flex-wrap: wrap;
+  width: 100%;
+  max-height: 305px;
+  overflow-x: hidden;
+  overflow-y: auto;
+  overscroll-behavior: none auto;
+  padding-bottom: 8px;
+  ${({ menuFonts }) => menuFonts}
+`;
+
+const List = styled.ul.attrs({ role: 'listbox' })`
+  width: 100%;
+  padding: 5px 0;
+  margin: 0;
+  font-size: 14px;
+  text-align: left;
+  list-style: none;
+  border-bottom: 1px solid ${({ theme }) => rgba(theme.colors.bg.v0, 0.1)};
+`;
+
+const Item = styled.li.attrs(({ fontFamily }) => ({
+  tabIndex: '0',
+  role: 'option',
+  style: {
+    fontFamily,
+  },
+}))`
+  letter-spacing: ${({ theme }) => theme.fonts.label.letterSpacing};
+  padding: 8px 12px;
+  margin: 0;
+  white-space: nowrap;
+  overflow: hidden;
+  font-size: ${({ theme }) => theme.fonts.label.size};
+  line-height: ${({ theme }) => theme.fonts.label.lineHeight};
+  font-weight: ${({ theme }) => theme.fonts.label.weight};
+
+  &:hover {
+    background-color: ${({ theme }) => theme.colors.bg.v12};
+  }
+
+  &:focus {
+    background-color: ${({ theme }) => theme.colors.bg.v12};
+    outline: none;
+  }
+`;
+
+const NoItem = styled.span`
+  letter-spacing: ${({ theme }) => theme.fonts.label.letterSpacing};
+  padding: 8px 12px 0 12px;
+  margin: 0;
+  font-style: italic;
+  color: ${({ theme }) => rgba(theme.colors.bg.v0, 0.54)};
+  font-size: ${({ theme }) => theme.fonts.body1.size};
+  line-height: ${({ theme }) => theme.fonts.body1.lineHeight};
+`;
+
+const BoxedTextInput = styled(TextInput)`
+  padding: 6px;
+  border: 1px solid ${({ theme }) => theme.colors.bg.v4};
+  border-radius: 4px;
+  background-color: ${({ theme }) => theme.colors.fg.v1};
+  color: ${({ theme }) => theme.colors.fg.v0};
+`;
+
+const ExpandedTextInput = styled(BoxedTextInput)`
+  flex-grow: 1;
+  margin: 8px;
+`;
+
+const LIST_PADDING = 5;
+const FONT_ROW_HEIGHT = 34;
+
+function FontPickerContainer({ handleCurrentValue, onClose }) {
+  const {
+    state: { fonts, recentUsedFontValues },
+    actions: { insertUsedFont, getMenuFonts },
+  } = useFont();
+
+  const pickerContainerRef = useRef();
+  const listContainerRef = useRef();
+  const listRef = useRef();
+  const inputRef = useRef();
+  const currentActiveRef = useRef(0);
+
+  const [currentIndex, setCurrentIndex] = useState(0);
+  const [searchValue, setSearchValue] = useState('');
+  const [menuFonts, setMenuFonts] = useState();
+
+  // Recently used font options, clean up on refresh
+  const recentUsedFonts = useMemo(() => {
+    const fontsFromSlug = recentUsedFontValues.map((value) =>
+      fonts.find((font) => font.value === value)
+    );
+    return searchValue && searchValue !== ''
```

Comment:
The filtering is the same all the time. Let's move the matcher as a top-level function and call it everywhere.

Reply:
Done